packages/account/src/Sections/Security/ClosingAccount/closing-account-reason.tsx

```diff
@@ -24,73 +37,33 @@ const preparingReason = values => {
     return selected_reasons.replace(/(\r\n|\n|\r)/gm, ' ');
 };
 
-const selectedReasons = values => {
+const selectedReasons = (values: FormikValues) => {
     return Object.entries(values).filter(
         ([key, value]) => !['other_trading_platforms', 'do_to_improve'].includes(key) && value
     );
 };
 
-const WarningModal = props => {
-    return (
-        <div className='account-closure-warning-modal'>
-            <Icon icon='IcRedWarning' size={96} />
-            <Text size='xs' line_height='x' weight='bold' className='account-closure-warning-modal__warning-message'>
-                {localize('Close your account?')}
-            </Text>
-            <div className='account-closure-warning-modal__content-wrapper'>
-                <Text as='p' align='center' className='account-closure-warning-modal__content'>
-                    {localize(
-                        'Closing your account will automatically log you out. We shall delete your personal information as soon as our legal obligations are met.'
-                    )}
-                </Text>
-            </div>
-            <FormSubmitButton
-                is_disabled={false}
-                label={localize('Close account')}
-                className='account-closure-warning-modal__close-account-button'
-                has_cancel
-                cancel_label={localize('Go Back')}
-                onClick={() => props.startDeactivating()}
-                onCancel={() => props.closeModal()}
-            />
-        </div>
-    );
-};
-
-const GeneralErrorContent = ({ message, onClick }) => (
-    <>
-        <div className='closing-account-error__container closing-account-error__container-message'>
-            <div className='closing-account-error__details closing-account-error__details-message'>{message}</div>
-        </div>
-        <div>
-            <Button className='closing-account-error__button' primary onClick={onClick}>
-                {localize('OK')}
-            </Button>
-        </div>
-    </>
-);
-
-const character_limit_no = 110;
-const max_allowed_reasons = 3;
-
-const ClosingAccountReason = observer(({ onBackClick }) => {
-    const { client } = useStore();
-    const { dxtrade_accounts_list, mt5_login_list, account_list } = client;
-    const { is_appstore } = React.useContext(PlatformContext);
+const ClosingAccountReason = ({ onBackClick }: TClosingAccountReasonProps) => {
     const [is_account_closed, setIsAccountClosed] = React.useState(false);
     const [is_loading, setIsLoading] = React.useState(false);
     const [is_modal_open, setIsModalOpen] = React.useState(false);
-    const [which_modal_should_render, setWhichModalShouldRender] = React.useState();
-    const [reason, setReason] = React.useState(null);
+    const [which_modal_should_render, setWhichModalShouldRender] = React.useState<string>('');
+    const [reason, setReason] = React.useState<string>('');
```

Comment:
Just a suggestion, we don't need to put the type where we're passing the initial value, typescript can infer from the initial value